tests/preprocessing/test_encode.py

```diff
@@ -42,11 +42,36 @@ def test_autodetect_encode(self):
         }
         assert id(encoded_ann_data.X) != id(encoded_ann_data.layers["original"])
 
+    def test_autodetect_num_only(self, capfd):
+        adata = read(dataset_path=f"{_TEST_PATH}/dataset2.csv")
+        encoded_ann_data = encode(adata, autodetect=True)
+        out, err = capfd.readouterr()
```

Comment:
I'd prefer a simple assert whether the error was thrown. No need to check the message. If we ever were to change the error message we would also have to change it here. Annoying.

Reply:
Would have done this normally, yes. The thing is, we do not throw an error here (we just print a warning).

Reply:
Ah okay.I get it. The message is ugly AF though :)

Should be more like:

"No columns to be encoded were detected" :)